// @mobsya/thymio-api/src/thymio_generated.ts
// automatically generated by the FlatBuffers compiler, do not modify

/**
 * The Type of a node
 * A node can designate either a robot, a simulator, each with various capabilities.
 *
 * @enum
 */
export namespace mobsya.fb{
export enum NodeType{
  Thymio2= 0,
  Thymio2Wireless= 1,
  SimulatedThymio2= 2,
  DummyNode= 3,
  UnknownType= 4
}};

/**
 * Capabilities of a node
 * What a client can do with a node may depends on permissions, firmware version, type of node/device, etc
 *
 * @enum
 */
export namespace mobsya.fb{
export enum NodeCapability{
  Rename= 1,
  ForceResetAndStop= 2
}};

/**
 * The status of a node represents whether the node is connected and available.
 *
 * @enum
 */
export namespace mobsya.fb{
export enum NodeStatus{
  /**
   * The status is unknown, this denotes a programmation error or a bug
   */
  unknown= 0,

  /**
   * The node is just connected. However, sending code or requesting a VM desciption is unspecified in this state
   */
  connected= 1,

  /**
   * The node is available. It needs to be sent a LockNode message before any action that would mutate it state, such as sending code can take place
   */
  available= 2,

  /**
   * The node is busy - this will be sent to all clients, except the one who locked the node
   */
  busy= 3,

  /**
   * The node is ready - this will be sent the client that optained a lock on the node
   */
  ready= 4,

  /**
   * The node is disconnected. Any reference to it will be destroyed on the server side, and sending further messages to it will fail
   */
  disconnected= 5
}};

/**
 * @enum
 */
export namespace mobsya.fb{
export enum ErrorType{
  no_error= 0,

  /**
   * Genric error
   */
  unknown_error= 1,

  /**
   * The node ddoes not exist or was disconnected
   */
  unknown_node= 2,

  /**
   * The node is busy (locked by another client)
   */
  node_busy= 3,

  /**
   *Unsuported variable type - For SetVariables and Events
   */
  unsupported_variable_type= 4
}};

/**
 * @enum
 */
export namespace mobsya.fb{
export enum WatchableInfo{
  StopMonitoring= 0,
  Variables= 1,
  Events= 2,
  VMExecutionState= 4,
  SharedEventsDescription= 8,
  SharedVariables= 16,
  Scratchpads= 32
}};

/**
 * @enum
 */
export namespace mobsya.fb{
export enum ProgrammingLanguage{
  Aseba= 1,
  Aesl= 2
}};

/**
 * @enum
 */
export namespace mobsya.fb{
export enum CompilationOptions{
  NoOption= 1,
  LoadOnTarget= 2,
  FetchBytecode= 4
}};

/**
 * @enum
 */
export namespace mobsya.fb{
export enum VMExecutionStateCommand{
  Stop= 0,
  Run= 1,
  Step= 2,
  StepToNextLine= 3,
  Pause= 4,
  Reset= 5,
  Reboot= 6,
  Suspend= 7,
  WriteProgramToDeviceMemory= 8
}};

/**
 * @enum
 */
export namespace mobsya.fb{
export enum VMExecutionState{
  Stopped= 0,
  Running= 1,
  Paused= 2
}};

/**
 * @enum
 */
export namespace mobsya.fb{
export enum VMExecutionError{
  NoError= 0,
  Killed= 1,
  OutOfBoundAccess= 2,
  DivisionByZero= 3,
  GenericError= 4
}};

/**
 * @enum
 */
export namespace mobsya.fb{
export enum AnyMessage{
  NONE= 0,
  ConnectionHandshake= 1,
  DeviceManagerShutdownRequest= 2,
  RequestListOfNodes= 3,
  RequestNodeAsebaVMDescription= 4,
  LockNode= 5,
  UnlockNode= 6,
  RenameNode= 7,
  CompileAndLoadCodeOnVM= 8,
  NodesChanged= 9,
  NodeAsebaVMDescription= 10,
  RequestCompleted= 11,
  Error= 12,
  CompilationResultFailure= 13,
  CompilationResultSuccess= 14,
  WatchNode= 15,
  VariablesChanged= 16,
  SetVariables= 17,
  EventsDescriptionsChanged= 18,
  RegisterEvents= 19,
  SendEvents= 20,
  EventsEmitted= 21,
  SetBreakpoints= 22,
  SetBreakpointsResponse= 23,
  SetVMExecutionState= 24,
  VMExecutionStateChanged= 25,
  ScratchpadUpdate= 26
}};

/**
 * @constructor
 */
export namespace mobsya.fb{
export class ConnectionHandshake {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns ConnectionHandshake
 */
__init(i:number, bb:flatbuffers.ByteBuffer):ConnectionHandshake {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param ConnectionHandshake= obj
 * @returns ConnectionHandshake
 */
static getRootAsConnectionHandshake(bb:flatbuffers.ByteBuffer, obj?:ConnectionHandshake):ConnectionHandshake {
  return (obj || new ConnectionHandshake).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
minProtocolVersion():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 1;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_minProtocolVersion(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
protocolVersion():number {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 1;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_protocolVersion(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 6);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
maxMessageSize():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 102400;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_maxMessageSize(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param number index
 * @returns number
 */
token(index: number):number|null {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? this.bb!.readUint8(this.bb!.__vector(this.bb_pos + offset) + index) : 0;
};

/**
 * @returns number
 */
tokenLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @returns Uint8Array
 */
tokenArray():Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? new Uint8Array(this.bb!.bytes().buffer, this.bb!.bytes().byteOffset + this.bb!.__vector(this.bb_pos + offset), this.bb!.__vector_len(this.bb_pos + offset)) : null;
};

/**
 * @returns boolean
 */
localhostPeer():boolean {
  var offset = this.bb!.__offset(this.bb_pos, 12);
  return offset ? !!this.bb!.readInt8(this.bb_pos + offset) : false;
};

/**
 * @param boolean value
 * @returns boolean
 */
mutate_localhostPeer(value:boolean):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 12);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt8(this.bb_pos + offset, +value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startConnectionHandshake(builder:flatbuffers.Builder) {
  builder.startObject(5);
};

/**
 * @param flatbuffers.Builder builder
 * @param number minProtocolVersion
 */
static addMinProtocolVersion(builder:flatbuffers.Builder, minProtocolVersion:number) {
  builder.addFieldInt32(0, minProtocolVersion, 1);
};

/**
 * @param flatbuffers.Builder builder
 * @param number protocolVersion
 */
static addProtocolVersion(builder:flatbuffers.Builder, protocolVersion:number) {
  builder.addFieldInt32(1, protocolVersion, 1);
};

/**
 * @param flatbuffers.Builder builder
 * @param number maxMessageSize
 */
static addMaxMessageSize(builder:flatbuffers.Builder, maxMessageSize:number) {
  builder.addFieldInt32(2, maxMessageSize, 102400);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset tokenOffset
 */
static addToken(builder:flatbuffers.Builder, tokenOffset:flatbuffers.Offset) {
  builder.addFieldOffset(3, tokenOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<number> data
 * @returns flatbuffers.Offset
 */
static createTokenVector(builder:flatbuffers.Builder, data:number[] | Uint8Array):flatbuffers.Offset {
  builder.startVector(1, data.length, 1);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addInt8(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startTokenVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(1, numElems, 1);
};

/**
 * @param flatbuffers.Builder builder
 * @param boolean localhostPeer
 */
static addLocalhostPeer(builder:flatbuffers.Builder, localhostPeer:boolean) {
  builder.addFieldInt8(4, +localhostPeer, +false);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endConnectionHandshake(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class DeviceManagerShutdownRequest {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns DeviceManagerShutdownRequest
 */
__init(i:number, bb:flatbuffers.ByteBuffer):DeviceManagerShutdownRequest {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param DeviceManagerShutdownRequest= obj
 * @returns DeviceManagerShutdownRequest
 */
static getRootAsDeviceManagerShutdownRequest(bb:flatbuffers.ByteBuffer, obj?:DeviceManagerShutdownRequest):DeviceManagerShutdownRequest {
  return (obj || new DeviceManagerShutdownRequest).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startDeviceManagerShutdownRequest(builder:flatbuffers.Builder) {
  builder.startObject(1);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endDeviceManagerShutdownRequest(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 *A node id
 *
 * @constructor
 */
export namespace mobsya.fb{
export class NodeId {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns NodeId
 */
__init(i:number, bb:flatbuffers.ByteBuffer):NodeId {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param NodeId= obj
 * @returns NodeId
 */
static getRootAsNodeId(bb:flatbuffers.ByteBuffer, obj?:NodeId):NodeId {
  return (obj || new NodeId).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @param number index
 * @returns number
 */
id(index: number):number|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint8(this.bb!.__vector(this.bb_pos + offset) + index) : 0;
};

/**
 * @returns number
 */
idLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @returns Uint8Array
 */
idArray():Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? new Uint8Array(this.bb!.bytes().buffer, this.bb!.bytes().byteOffset + this.bb!.__vector(this.bb_pos + offset), this.bb!.__vector_len(this.bb_pos + offset)) : null;
};

/**
 * @param flatbuffers.Builder builder
 */
static startNodeId(builder:flatbuffers.Builder) {
  builder.startObject(1);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset idOffset
 */
static addId(builder:flatbuffers.Builder, idOffset:flatbuffers.Offset) {
  builder.addFieldOffset(0, idOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<number> data
 * @returns flatbuffers.Offset
 */
static createIdVector(builder:flatbuffers.Builder, data:number[] | Uint8Array):flatbuffers.Offset {
  builder.startVector(1, data.length, 1);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addInt8(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startIdVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(1, numElems, 1);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endNodeId(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * Represent a node
 *
 * @constructor
 */
export namespace mobsya.fb{
export class Node {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns Node
 */
__init(i:number, bb:flatbuffers.ByteBuffer):Node {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param Node= obj
 * @returns Node
 */
static getRootAsNode(bb:flatbuffers.ByteBuffer, obj?:Node):Node {
  return (obj || new Node).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
groupId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @returns mobsya.fb.NodeStatus
 */
status():mobsya.fb.NodeStatus {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? /**  */ (this.bb!.readInt32(this.bb_pos + offset)) : mobsya.fb.NodeStatus.unknown;
};

/**
 * @param mobsya.fb.NodeStatus value
 * @returns boolean
 */
mutate_status(value:mobsya.fb.NodeStatus):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns mobsya.fb.NodeType
 */
type():mobsya.fb.NodeType {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? /**  */ (this.bb!.readInt32(this.bb_pos + offset)) : mobsya.fb.NodeType.Thymio2;
};

/**
 * @param mobsya.fb.NodeType value
 * @returns boolean
 */
mutate_type(value:mobsya.fb.NodeType):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 10);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
name():string|null
name(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
name(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 12);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @returns flatbuffers.Long
 */
capabilities():flatbuffers.Long {
  var offset = this.bb!.__offset(this.bb_pos, 14);
  return offset ? this.bb!.readUint64(this.bb_pos + offset) : this.bb!.createLong(0, 0);
};

/**
 * @param flatbuffers.Long value
 * @returns boolean
 */
mutate_capabilities(value:flatbuffers.Long):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 14);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint64(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
fwVersion():string|null
fwVersion(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
fwVersion(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 16);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
latestFwVersion():string|null
latestFwVersion(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
latestFwVersion(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 18);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @param flatbuffers.Builder builder
 */
static startNode(builder:flatbuffers.Builder) {
  builder.startObject(8);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(0, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset groupIdOffset
 */
static addGroupId(builder:flatbuffers.Builder, groupIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, groupIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param mobsya.fb.NodeStatus status
 */
static addStatus(builder:flatbuffers.Builder, status:mobsya.fb.NodeStatus) {
  builder.addFieldInt32(2, status, mobsya.fb.NodeStatus.unknown);
};

/**
 * @param flatbuffers.Builder builder
 * @param mobsya.fb.NodeType type
 */
static addType(builder:flatbuffers.Builder, type:mobsya.fb.NodeType) {
  builder.addFieldInt32(3, type, mobsya.fb.NodeType.Thymio2);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nameOffset
 */
static addName(builder:flatbuffers.Builder, nameOffset:flatbuffers.Offset) {
  builder.addFieldOffset(4, nameOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Long capabilities
 */
static addCapabilities(builder:flatbuffers.Builder, capabilities:flatbuffers.Long) {
  builder.addFieldInt64(5, capabilities, builder.createLong(0, 0));
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset fwVersionOffset
 */
static addFwVersion(builder:flatbuffers.Builder, fwVersionOffset:flatbuffers.Offset) {
  builder.addFieldOffset(6, fwVersionOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset latestFwVersionOffset
 */
static addLatestFwVersion(builder:flatbuffers.Builder, latestFwVersionOffset:flatbuffers.Offset) {
  builder.addFieldOffset(7, latestFwVersionOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endNode(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class NamedValue {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns NamedValue
 */
__init(i:number, bb:flatbuffers.ByteBuffer):NamedValue {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param NamedValue= obj
 * @returns NamedValue
 */
static getRootAsNamedValue(bb:flatbuffers.ByteBuffer, obj?:NamedValue):NamedValue {
  return (obj || new NamedValue).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
name():string|null
name(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
name(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @param number index
 * @returns number
 */
value(index: number):number|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.readUint8(this.bb!.__vector(this.bb_pos + offset) + index) : 0;
};

/**
 * @returns number
 */
valueLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @returns Uint8Array
 */
valueArray():Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? new Uint8Array(this.bb!.bytes().buffer, this.bb!.bytes().byteOffset + this.bb!.__vector(this.bb_pos + offset), this.bb!.__vector_len(this.bb_pos + offset)) : null;
};

/**
 * @param flatbuffers.Builder builder
 */
static startNamedValue(builder:flatbuffers.Builder) {
  builder.startObject(2);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nameOffset
 */
static addName(builder:flatbuffers.Builder, nameOffset:flatbuffers.Offset) {
  builder.addFieldOffset(0, nameOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset valueOffset
 */
static addValue(builder:flatbuffers.Builder, valueOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, valueOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<number> data
 * @returns flatbuffers.Offset
 */
static createValueVector(builder:flatbuffers.Builder, data:number[] | Uint8Array):flatbuffers.Offset {
  builder.startVector(1, data.length, 1);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addInt8(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startValueVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(1, numElems, 1);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endNamedValue(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  builder.requiredField(offset, 4); // name
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class NodeVariable {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns NodeVariable
 */
__init(i:number, bb:flatbuffers.ByteBuffer):NodeVariable {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param NodeVariable= obj
 * @returns NodeVariable
 */
static getRootAsNodeVariable(bb:flatbuffers.ByteBuffer, obj?:NodeVariable):NodeVariable {
  return (obj || new NodeVariable).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
name():string|null
name(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
name(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @param number index
 * @returns number
 */
value(index: number):number|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.readUint8(this.bb!.__vector(this.bb_pos + offset) + index) : 0;
};

/**
 * @returns number
 */
valueLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @returns Uint8Array
 */
valueArray():Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? new Uint8Array(this.bb!.bytes().buffer, this.bb!.bytes().byteOffset + this.bb!.__vector(this.bb_pos + offset), this.bb!.__vector_len(this.bb_pos + offset)) : null;
};

/**
 * @param flatbuffers.Builder builder
 */
static startNodeVariable(builder:flatbuffers.Builder) {
  builder.startObject(2);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nameOffset
 */
static addName(builder:flatbuffers.Builder, nameOffset:flatbuffers.Offset) {
  builder.addFieldOffset(0, nameOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset valueOffset
 */
static addValue(builder:flatbuffers.Builder, valueOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, valueOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<number> data
 * @returns flatbuffers.Offset
 */
static createValueVector(builder:flatbuffers.Builder, data:number[] | Uint8Array):flatbuffers.Offset {
  builder.startVector(1, data.length, 1);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addInt8(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startValueVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(1, numElems, 1);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endNodeVariable(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  builder.requiredField(offset, 4); // name
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class VariablesChanged {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns VariablesChanged
 */
__init(i:number, bb:flatbuffers.ByteBuffer):VariablesChanged {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param VariablesChanged= obj
 * @returns VariablesChanged
 */
static getRootAsVariablesChanged(bb:flatbuffers.ByteBuffer, obj?:VariablesChanged):VariablesChanged {
  return (obj || new VariablesChanged).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param number index
 * @param mobsya.fb.NodeVariable= obj
 * @returns mobsya.fb.NodeVariable
 */
vars(index: number, obj?:mobsya.fb.NodeVariable):mobsya.fb.NodeVariable|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeVariable).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
varsLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * unix timestamp ( miliseconds) of when the variables were modified - or 0 if unspecified
 *
 * @returns flatbuffers.Long
 */
timestamp():flatbuffers.Long {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.readUint64(this.bb_pos + offset) : this.bb!.createLong(0, 0);
};

/**
 * @param flatbuffers.Long value
 * @returns boolean
 */
mutate_timestamp(value:flatbuffers.Long):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint64(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startVariablesChanged(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(0, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset varsOffset
 */
static addVars(builder:flatbuffers.Builder, varsOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, varsOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createVarsVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startVarsVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Long timestamp
 */
static addTimestamp(builder:flatbuffers.Builder, timestamp:flatbuffers.Long) {
  builder.addFieldInt64(2, timestamp, builder.createLong(0, 0));
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endVariablesChanged(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class SendEvents {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns SendEvents
 */
__init(i:number, bb:flatbuffers.ByteBuffer):SendEvents {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param SendEvents= obj
 * @returns SendEvents
 */
static getRootAsSendEvents(bb:flatbuffers.ByteBuffer, obj?:SendEvents):SendEvents {
  return (obj || new SendEvents).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param number index
 * @param mobsya.fb.NamedValue= obj
 * @returns mobsya.fb.NamedValue
 */
events(index: number, obj?:mobsya.fb.NamedValue):mobsya.fb.NamedValue|null {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? (obj || new mobsya.fb.NamedValue).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
eventsLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @param flatbuffers.Builder builder
 */
static startSendEvents(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset eventsOffset
 */
static addEvents(builder:flatbuffers.Builder, eventsOffset:flatbuffers.Offset) {
  builder.addFieldOffset(2, eventsOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createEventsVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startEventsVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endSendEvents(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class EventsEmitted {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns EventsEmitted
 */
__init(i:number, bb:flatbuffers.ByteBuffer):EventsEmitted {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param EventsEmitted= obj
 * @returns EventsEmitted
 */
static getRootAsEventsEmitted(bb:flatbuffers.ByteBuffer, obj?:EventsEmitted):EventsEmitted {
  return (obj || new EventsEmitted).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param number index
 * @param mobsya.fb.NamedValue= obj
 * @returns mobsya.fb.NamedValue
 */
events(index: number, obj?:mobsya.fb.NamedValue):mobsya.fb.NamedValue|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NamedValue).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
eventsLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * unix timestamp ( miliseconds) of when the events were emitted - or 0 if unspecified
 *
 * @returns flatbuffers.Long
 */
timestamp():flatbuffers.Long {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.readUint64(this.bb_pos + offset) : this.bb!.createLong(0, 0);
};

/**
 * @param flatbuffers.Long value
 * @returns boolean
 */
mutate_timestamp(value:flatbuffers.Long):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint64(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startEventsEmitted(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(0, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset eventsOffset
 */
static addEvents(builder:flatbuffers.Builder, eventsOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, eventsOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createEventsVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startEventsVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Long timestamp
 */
static addTimestamp(builder:flatbuffers.Builder, timestamp:flatbuffers.Long) {
  builder.addFieldInt64(2, timestamp, builder.createLong(0, 0));
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endEventsEmitted(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * Message sent by the server when a node changes status.
 * This messages is automatically sent when the application first connects and each time one or more node change status
 *
 * @constructor
 */
export namespace mobsya.fb{
export class NodesChanged {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns NodesChanged
 */
__init(i:number, bb:flatbuffers.ByteBuffer):NodesChanged {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param NodesChanged= obj
 * @returns NodesChanged
 */
static getRootAsNodesChanged(bb:flatbuffers.ByteBuffer, obj?:NodesChanged):NodesChanged {
  return (obj || new NodesChanged).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * Nodes whose status changed. it's up to the client to maintain a list of all nodes.
 *
 * @param number index
 * @param mobsya.fb.Node= obj
 * @returns mobsya.fb.Node
 */
nodes(index: number, obj?:mobsya.fb.Node):mobsya.fb.Node|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? (obj || new mobsya.fb.Node).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
nodesLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @param flatbuffers.Builder builder
 */
static startNodesChanged(builder:flatbuffers.Builder) {
  builder.startObject(1);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodesOffset
 */
static addNodes(builder:flatbuffers.Builder, nodesOffset:flatbuffers.Offset) {
  builder.addFieldOffset(0, nodesOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createNodesVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startNodesVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endNodesChanged(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * Ask the server to send a list of all nodes
 * The server will send back a NodesChanged message containing infos on all nodes
 *
 * @constructor
 */
export namespace mobsya.fb{
export class RequestListOfNodes {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns RequestListOfNodes
 */
__init(i:number, bb:flatbuffers.ByteBuffer):RequestListOfNodes {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param RequestListOfNodes= obj
 * @returns RequestListOfNodes
 */
static getRootAsRequestListOfNodes(bb:flatbuffers.ByteBuffer, obj?:RequestListOfNodes):RequestListOfNodes {
  return (obj || new RequestListOfNodes).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @param flatbuffers.Builder builder
 */
static startRequestListOfNodes(builder:flatbuffers.Builder) {
  builder.startObject(0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endRequestListOfNodes(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * Request the server to send a NodeAsebaVMDescription for the node corresponding to node_id
 *
 * @constructor
 */
export namespace mobsya.fb{
export class RequestNodeAsebaVMDescription {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns RequestNodeAsebaVMDescription
 */
__init(i:number, bb:flatbuffers.ByteBuffer):RequestNodeAsebaVMDescription {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param RequestNodeAsebaVMDescription= obj
 * @returns RequestNodeAsebaVMDescription
 */
static getRootAsRequestNodeAsebaVMDescription(bb:flatbuffers.ByteBuffer, obj?:RequestNodeAsebaVMDescription):RequestNodeAsebaVMDescription {
  return (obj || new RequestNodeAsebaVMDescription).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param flatbuffers.Builder builder
 */
static startRequestNodeAsebaVMDescription(builder:flatbuffers.Builder) {
  builder.startObject(2);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endRequestNodeAsebaVMDescription(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * The description of an Aseba VM, sent in response to a RequestNodeAsebaVMDescription
 *
 * @constructor
 */
export namespace mobsya.fb{
export class NodeAsebaVMDescription {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns NodeAsebaVMDescription
 */
__init(i:number, bb:flatbuffers.ByteBuffer):NodeAsebaVMDescription {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param NodeAsebaVMDescription= obj
 * @returns NodeAsebaVMDescription
 */
static getRootAsNodeAsebaVMDescription(bb:flatbuffers.ByteBuffer, obj?:NodeAsebaVMDescription):NodeAsebaVMDescription {
  return (obj || new NodeAsebaVMDescription).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @returns number
 */
bytecodeSize():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_bytecode_size(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
dataSize():number {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_data_size(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 10);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
stackSize():number {
  var offset = this.bb!.__offset(this.bb_pos, 12);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_stack_size(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 12);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param number index
 * @param mobsya.fb.AsebaNamedVariable= obj
 * @returns mobsya.fb.AsebaNamedVariable
 */
variables(index: number, obj?:mobsya.fb.AsebaNamedVariable):mobsya.fb.AsebaNamedVariable|null {
  var offset = this.bb!.__offset(this.bb_pos, 14);
  return offset ? (obj || new mobsya.fb.AsebaNamedVariable).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
variablesLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 14);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @param number index
 * @param mobsya.fb.AsebaEvent= obj
 * @returns mobsya.fb.AsebaEvent
 */
events(index: number, obj?:mobsya.fb.AsebaEvent):mobsya.fb.AsebaEvent|null {
  var offset = this.bb!.__offset(this.bb_pos, 16);
  return offset ? (obj || new mobsya.fb.AsebaEvent).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
eventsLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 16);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @param number index
 * @param mobsya.fb.AsebaNativeFunction= obj
 * @returns mobsya.fb.AsebaNativeFunction
 */
functions(index: number, obj?:mobsya.fb.AsebaNativeFunction):mobsya.fb.AsebaNativeFunction|null {
  var offset = this.bb!.__offset(this.bb_pos, 18);
  return offset ? (obj || new mobsya.fb.AsebaNativeFunction).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
functionsLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 18);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @param flatbuffers.Builder builder
 */
static startNodeAsebaVMDescription(builder:flatbuffers.Builder) {
  builder.startObject(8);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number bytecodeSize
 */
static addBytecodeSize(builder:flatbuffers.Builder, bytecodeSize:number) {
  builder.addFieldInt32(2, bytecodeSize, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number dataSize
 */
static addDataSize(builder:flatbuffers.Builder, dataSize:number) {
  builder.addFieldInt32(3, dataSize, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number stackSize
 */
static addStackSize(builder:flatbuffers.Builder, stackSize:number) {
  builder.addFieldInt32(4, stackSize, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset variablesOffset
 */
static addVariables(builder:flatbuffers.Builder, variablesOffset:flatbuffers.Offset) {
  builder.addFieldOffset(5, variablesOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createVariablesVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startVariablesVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset eventsOffset
 */
static addEvents(builder:flatbuffers.Builder, eventsOffset:flatbuffers.Offset) {
  builder.addFieldOffset(6, eventsOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createEventsVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startEventsVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset functionsOffset
 */
static addFunctions(builder:flatbuffers.Builder, functionsOffset:flatbuffers.Offset) {
  builder.addFieldOffset(7, functionsOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createFunctionsVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startFunctionsVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endNodeAsebaVMDescription(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class AsebaNamedVariable {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns AsebaNamedVariable
 */
__init(i:number, bb:flatbuffers.ByteBuffer):AsebaNamedVariable {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param AsebaNamedVariable= obj
 * @returns AsebaNamedVariable
 */
static getRootAsAsebaNamedVariable(bb:flatbuffers.ByteBuffer, obj?:AsebaNamedVariable):AsebaNamedVariable {
  return (obj || new AsebaNamedVariable).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
index():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint16(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_index(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint16(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
name():string|null
name(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
name(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @returns number
 */
size():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_size(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startAsebaNamedVariable(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param number index
 */
static addIndex(builder:flatbuffers.Builder, index:number) {
  builder.addFieldInt16(0, index, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nameOffset
 */
static addName(builder:flatbuffers.Builder, nameOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nameOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number size
 */
static addSize(builder:flatbuffers.Builder, size:number) {
  builder.addFieldInt32(2, size, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endAsebaNamedVariable(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class AsebaEvent {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns AsebaEvent
 */
__init(i:number, bb:flatbuffers.ByteBuffer):AsebaEvent {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param AsebaEvent= obj
 * @returns AsebaEvent
 */
static getRootAsAsebaEvent(bb:flatbuffers.ByteBuffer, obj?:AsebaEvent):AsebaEvent {
  return (obj || new AsebaEvent).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
index():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint16(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_index(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint16(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
name():string|null
name(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
name(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
description():string|null
description(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
description(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @param flatbuffers.Builder builder
 */
static startAsebaEvent(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param number index
 */
static addIndex(builder:flatbuffers.Builder, index:number) {
  builder.addFieldInt16(0, index, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nameOffset
 */
static addName(builder:flatbuffers.Builder, nameOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nameOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset descriptionOffset
 */
static addDescription(builder:flatbuffers.Builder, descriptionOffset:flatbuffers.Offset) {
  builder.addFieldOffset(2, descriptionOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endAsebaEvent(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class AsebaNativeFunction {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns AsebaNativeFunction
 */
__init(i:number, bb:flatbuffers.ByteBuffer):AsebaNativeFunction {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param AsebaNativeFunction= obj
 * @returns AsebaNativeFunction
 */
static getRootAsAsebaNativeFunction(bb:flatbuffers.ByteBuffer, obj?:AsebaNativeFunction):AsebaNativeFunction {
  return (obj || new AsebaNativeFunction).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
index():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint16(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_index(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint16(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
name():string|null
name(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
name(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
description():string|null
description(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
description(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @param number index
 * @param mobsya.fb.AsebaNativeFunctionParameter= obj
 * @returns mobsya.fb.AsebaNativeFunctionParameter
 */
parameters(index: number, obj?:mobsya.fb.AsebaNativeFunctionParameter):mobsya.fb.AsebaNativeFunctionParameter|null {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? (obj || new mobsya.fb.AsebaNativeFunctionParameter).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
parametersLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @param flatbuffers.Builder builder
 */
static startAsebaNativeFunction(builder:flatbuffers.Builder) {
  builder.startObject(4);
};

/**
 * @param flatbuffers.Builder builder
 * @param number index
 */
static addIndex(builder:flatbuffers.Builder, index:number) {
  builder.addFieldInt16(0, index, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nameOffset
 */
static addName(builder:flatbuffers.Builder, nameOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nameOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset descriptionOffset
 */
static addDescription(builder:flatbuffers.Builder, descriptionOffset:flatbuffers.Offset) {
  builder.addFieldOffset(2, descriptionOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset parametersOffset
 */
static addParameters(builder:flatbuffers.Builder, parametersOffset:flatbuffers.Offset) {
  builder.addFieldOffset(3, parametersOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createParametersVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startParametersVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endAsebaNativeFunction(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class AsebaNativeFunctionParameter {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns AsebaNativeFunctionParameter
 */
__init(i:number, bb:flatbuffers.ByteBuffer):AsebaNativeFunctionParameter {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param AsebaNativeFunctionParameter= obj
 * @returns AsebaNativeFunctionParameter
 */
static getRootAsAsebaNativeFunctionParameter(bb:flatbuffers.ByteBuffer, obj?:AsebaNativeFunctionParameter):AsebaNativeFunctionParameter {
  return (obj || new AsebaNativeFunctionParameter).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
name():string|null
name(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
name(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @returns number
 */
size():number {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_size(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 6);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startAsebaNativeFunctionParameter(builder:flatbuffers.Builder) {
  builder.startObject(2);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nameOffset
 */
static addName(builder:flatbuffers.Builder, nameOffset:flatbuffers.Offset) {
  builder.addFieldOffset(0, nameOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number size
 */
static addSize(builder:flatbuffers.Builder, size:number) {
  builder.addFieldInt32(1, size, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endAsebaNativeFunctionParameter(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * Modify the value of specified variables
 *
 * @constructor
 */
export namespace mobsya.fb{
export class SetVariables {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns SetVariables
 */
__init(i:number, bb:flatbuffers.ByteBuffer):SetVariables {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param SetVariables= obj
 * @returns SetVariables
 */
static getRootAsSetVariables(bb:flatbuffers.ByteBuffer, obj?:SetVariables):SetVariables {
  return (obj || new SetVariables).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeOrGroupId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param number index
 * @param mobsya.fb.NodeVariable= obj
 * @returns mobsya.fb.NodeVariable
 */
vars(index: number, obj?:mobsya.fb.NodeVariable):mobsya.fb.NodeVariable|null {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? (obj || new mobsya.fb.NodeVariable).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
varsLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @param flatbuffers.Builder builder
 */
static startSetVariables(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeOrGroupIdOffset
 */
static addNodeOrGroupId(builder:flatbuffers.Builder, nodeOrGroupIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeOrGroupIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset varsOffset
 */
static addVars(builder:flatbuffers.Builder, varsOffset:flatbuffers.Offset) {
  builder.addFieldOffset(2, varsOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createVarsVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startVarsVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endSetVariables(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * Set a new name for this node
 *
 * @constructor
 */
export namespace mobsya.fb{
export class RenameNode {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns RenameNode
 */
__init(i:number, bb:flatbuffers.ByteBuffer):RenameNode {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param RenameNode= obj
 * @returns RenameNode
 */
static getRootAsRenameNode(bb:flatbuffers.ByteBuffer, obj?:RenameNode):RenameNode {
  return (obj || new RenameNode).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
newName():string|null
newName(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
newName(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @param flatbuffers.Builder builder
 */
static startRenameNode(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset newNameOffset
 */
static addNewName(builder:flatbuffers.Builder, newNameOffset:flatbuffers.Offset) {
  builder.addFieldOffset(2, newNameOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endRenameNode(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * Ask the server to give this client sole ownership of the node.
 * This needs to be done once before sending mutating instructions to the node
 * The lock remain until the client disconnect, the device disconnect or a UnlockNode message is sent
 * Once locked, the node appears busy to all clients.
 *
 * @constructor
 */
export namespace mobsya.fb{
export class LockNode {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns LockNode
 */
__init(i:number, bb:flatbuffers.ByteBuffer):LockNode {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param LockNode= obj
 * @returns LockNode
 */
static getRootAsLockNode(bb:flatbuffers.ByteBuffer, obj?:LockNode):LockNode {
  return (obj || new LockNode).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param flatbuffers.Builder builder
 */
static startLockNode(builder:flatbuffers.Builder) {
  builder.startObject(2);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endLockNode(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * Unlock a device so that other client can use it.
 * Once unlocked, the node appears ready to all clients.
 *
 * @constructor
 */
export namespace mobsya.fb{
export class UnlockNode {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns UnlockNode
 */
__init(i:number, bb:flatbuffers.ByteBuffer):UnlockNode {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param UnlockNode= obj
 * @returns UnlockNode
 */
static getRootAsUnlockNode(bb:flatbuffers.ByteBuffer, obj?:UnlockNode):UnlockNode {
  return (obj || new UnlockNode).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param flatbuffers.Builder builder
 */
static startUnlockNode(builder:flatbuffers.Builder) {
  builder.startObject(2);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endUnlockNode(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class CompileAndLoadCodeOnVM {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns CompileAndLoadCodeOnVM
 */
__init(i:number, bb:flatbuffers.ByteBuffer):CompileAndLoadCodeOnVM {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param CompileAndLoadCodeOnVM= obj
 * @returns CompileAndLoadCodeOnVM
 */
static getRootAsCompileAndLoadCodeOnVM(bb:flatbuffers.ByteBuffer, obj?:CompileAndLoadCodeOnVM):CompileAndLoadCodeOnVM {
  return (obj || new CompileAndLoadCodeOnVM).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @returns mobsya.fb.ProgrammingLanguage
 */
language():mobsya.fb.ProgrammingLanguage {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? /**  */ (this.bb!.readInt32(this.bb_pos + offset)) : mobsya.fb.ProgrammingLanguage.Aseba;
};

/**
 * @param mobsya.fb.ProgrammingLanguage value
 * @returns boolean
 */
mutate_language(value:mobsya.fb.ProgrammingLanguage):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
program():string|null
program(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
program(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @returns mobsya.fb.CompilationOptions
 */
options():mobsya.fb.CompilationOptions {
  var offset = this.bb!.__offset(this.bb_pos, 12);
  return offset ? /**  */ (this.bb!.readInt32(this.bb_pos + offset)) : /** } */ (0);
};

/**
 * @param mobsya.fb.CompilationOptions value
 * @returns boolean
 */
mutate_options(value:mobsya.fb.CompilationOptions):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 12);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startCompileAndLoadCodeOnVM(builder:flatbuffers.Builder) {
  builder.startObject(5);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param mobsya.fb.ProgrammingLanguage language
 */
static addLanguage(builder:flatbuffers.Builder, language:mobsya.fb.ProgrammingLanguage) {
  builder.addFieldInt32(2, language, mobsya.fb.ProgrammingLanguage.Aseba);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset programOffset
 */
static addProgram(builder:flatbuffers.Builder, programOffset:flatbuffers.Offset) {
  builder.addFieldOffset(3, programOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param mobsya.fb.CompilationOptions options
 */
static addOptions(builder:flatbuffers.Builder, options:mobsya.fb.CompilationOptions) {
  builder.addFieldInt32(4, options, /** } */ (0));
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endCompileAndLoadCodeOnVM(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  builder.requiredField(offset, 10); // program
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class CompilationResultFailure {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns CompilationResultFailure
 */
__init(i:number, bb:flatbuffers.ByteBuffer):CompilationResultFailure {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param CompilationResultFailure= obj
 * @returns CompilationResultFailure
 */
static getRootAsCompilationResultFailure(bb:flatbuffers.ByteBuffer, obj?:CompilationResultFailure):CompilationResultFailure {
  return (obj || new CompilationResultFailure).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
message():string|null
message(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
message(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @returns number
 */
character():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_character(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
line():number {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_line(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 10);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
column():number {
  var offset = this.bb!.__offset(this.bb_pos, 12);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_column(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 12);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startCompilationResultFailure(builder:flatbuffers.Builder) {
  builder.startObject(5);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset messageOffset
 */
static addMessage(builder:flatbuffers.Builder, messageOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, messageOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number character
 */
static addCharacter(builder:flatbuffers.Builder, character:number) {
  builder.addFieldInt32(2, character, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number line
 */
static addLine(builder:flatbuffers.Builder, line:number) {
  builder.addFieldInt32(3, line, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number column
 */
static addColumn(builder:flatbuffers.Builder, column:number) {
  builder.addFieldInt32(4, column, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endCompilationResultFailure(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  builder.requiredField(offset, 6); // message
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class CompilationResultSuccess {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns CompilationResultSuccess
 */
__init(i:number, bb:flatbuffers.ByteBuffer):CompilationResultSuccess {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param CompilationResultSuccess= obj
 * @returns CompilationResultSuccess
 */
static getRootAsCompilationResultSuccess(bb:flatbuffers.ByteBuffer, obj?:CompilationResultSuccess):CompilationResultSuccess {
  return (obj || new CompilationResultSuccess).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
bytecodeSize():number {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_bytecode_size(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 6);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
totalBytecodeSize():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_total_bytecode_size(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
variablesSize():number {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_variables_size(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 10);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
totalVariablesSize():number {
  var offset = this.bb!.__offset(this.bb_pos, 12);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_total_variables_size(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 12);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startCompilationResultSuccess(builder:flatbuffers.Builder) {
  builder.startObject(5);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number bytecodeSize
 */
static addBytecodeSize(builder:flatbuffers.Builder, bytecodeSize:number) {
  builder.addFieldInt32(1, bytecodeSize, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number totalBytecodeSize
 */
static addTotalBytecodeSize(builder:flatbuffers.Builder, totalBytecodeSize:number) {
  builder.addFieldInt32(2, totalBytecodeSize, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number variablesSize
 */
static addVariablesSize(builder:flatbuffers.Builder, variablesSize:number) {
  builder.addFieldInt32(3, variablesSize, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number totalVariablesSize
 */
static addTotalVariablesSize(builder:flatbuffers.Builder, totalVariablesSize:number) {
  builder.addFieldInt32(4, totalVariablesSize, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endCompilationResultSuccess(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class SetVMExecutionState {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns SetVMExecutionState
 */
__init(i:number, bb:flatbuffers.ByteBuffer):SetVMExecutionState {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param SetVMExecutionState= obj
 * @returns SetVMExecutionState
 */
static getRootAsSetVMExecutionState(bb:flatbuffers.ByteBuffer, obj?:SetVMExecutionState):SetVMExecutionState {
  return (obj || new SetVMExecutionState).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @returns mobsya.fb.VMExecutionStateCommand
 */
command():mobsya.fb.VMExecutionStateCommand {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? /**  */ (this.bb!.readInt16(this.bb_pos + offset)) : mobsya.fb.VMExecutionStateCommand.Stop;
};

/**
 * @param mobsya.fb.VMExecutionStateCommand value
 * @returns boolean
 */
mutate_command(value:mobsya.fb.VMExecutionStateCommand):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt16(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startSetVMExecutionState(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param mobsya.fb.VMExecutionStateCommand command
 */
static addCommand(builder:flatbuffers.Builder, command:mobsya.fb.VMExecutionStateCommand) {
  builder.addFieldInt16(2, command, mobsya.fb.VMExecutionStateCommand.Stop);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endSetVMExecutionState(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class VMExecutionStateChanged {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns VMExecutionStateChanged
 */
__init(i:number, bb:flatbuffers.ByteBuffer):VMExecutionStateChanged {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param VMExecutionStateChanged= obj
 * @returns VMExecutionStateChanged
 */
static getRootAsVMExecutionStateChanged(bb:flatbuffers.ByteBuffer, obj?:VMExecutionStateChanged):VMExecutionStateChanged {
  return (obj || new VMExecutionStateChanged).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @returns mobsya.fb.VMExecutionState
 */
state():mobsya.fb.VMExecutionState {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? /**  */ (this.bb!.readInt16(this.bb_pos + offset)) : mobsya.fb.VMExecutionState.Stopped;
};

/**
 * @param mobsya.fb.VMExecutionState value
 * @returns boolean
 */
mutate_state(value:mobsya.fb.VMExecutionState):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 6);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt16(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
line():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_line(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns mobsya.fb.VMExecutionError
 */
error():mobsya.fb.VMExecutionError {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? /**  */ (this.bb!.readInt16(this.bb_pos + offset)) : mobsya.fb.VMExecutionError.NoError;
};

/**
 * @param mobsya.fb.VMExecutionError value
 * @returns boolean
 */
mutate_error(value:mobsya.fb.VMExecutionError):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 10);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt16(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
errorMsg():string|null
errorMsg(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
errorMsg(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 12);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @param flatbuffers.Builder builder
 */
static startVMExecutionStateChanged(builder:flatbuffers.Builder) {
  builder.startObject(5);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(0, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param mobsya.fb.VMExecutionState state
 */
static addState(builder:flatbuffers.Builder, state:mobsya.fb.VMExecutionState) {
  builder.addFieldInt16(1, state, mobsya.fb.VMExecutionState.Stopped);
};

/**
 * @param flatbuffers.Builder builder
 * @param number line
 */
static addLine(builder:flatbuffers.Builder, line:number) {
  builder.addFieldInt32(2, line, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param mobsya.fb.VMExecutionError error
 */
static addError(builder:flatbuffers.Builder, error:mobsya.fb.VMExecutionError) {
  builder.addFieldInt16(3, error, mobsya.fb.VMExecutionError.NoError);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset errorMsgOffset
 */
static addErrorMsg(builder:flatbuffers.Builder, errorMsgOffset:flatbuffers.Offset) {
  builder.addFieldOffset(4, errorMsgOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endVMExecutionStateChanged(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class Breakpoint {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns Breakpoint
 */
__init(i:number, bb:flatbuffers.ByteBuffer):Breakpoint {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param Breakpoint= obj
 * @returns Breakpoint
 */
static getRootAsBreakpoint(bb:flatbuffers.ByteBuffer, obj?:Breakpoint):Breakpoint {
  return (obj || new Breakpoint).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
line():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_line(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startBreakpoint(builder:flatbuffers.Builder) {
  builder.startObject(1);
};

/**
 * @param flatbuffers.Builder builder
 * @param number line
 */
static addLine(builder:flatbuffers.Builder, line:number) {
  builder.addFieldInt32(0, line, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endBreakpoint(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class SetBreakpoints {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns SetBreakpoints
 */
__init(i:number, bb:flatbuffers.ByteBuffer):SetBreakpoints {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param SetBreakpoints= obj
 * @returns SetBreakpoints
 */
static getRootAsSetBreakpoints(bb:flatbuffers.ByteBuffer, obj?:SetBreakpoints):SetBreakpoints {
  return (obj || new SetBreakpoints).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param number index
 * @param mobsya.fb.Breakpoint= obj
 * @returns mobsya.fb.Breakpoint
 */
breakpoints(index: number, obj?:mobsya.fb.Breakpoint):mobsya.fb.Breakpoint|null {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? (obj || new mobsya.fb.Breakpoint).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
breakpointsLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @param flatbuffers.Builder builder
 */
static startSetBreakpoints(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset breakpointsOffset
 */
static addBreakpoints(builder:flatbuffers.Builder, breakpointsOffset:flatbuffers.Offset) {
  builder.addFieldOffset(2, breakpointsOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createBreakpointsVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startBreakpointsVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endSetBreakpoints(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class SetBreakpointsResponse {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns SetBreakpointsResponse
 */
__init(i:number, bb:flatbuffers.ByteBuffer):SetBreakpointsResponse {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param SetBreakpointsResponse= obj
 * @returns SetBreakpointsResponse
 */
static getRootAsSetBreakpointsResponse(bb:flatbuffers.ByteBuffer, obj?:SetBreakpointsResponse):SetBreakpointsResponse {
  return (obj || new SetBreakpointsResponse).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns mobsya.fb.ErrorType
 */
error():mobsya.fb.ErrorType {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? /**  */ (this.bb!.readInt32(this.bb_pos + offset)) : mobsya.fb.ErrorType.no_error;
};

/**
 * @param mobsya.fb.ErrorType value
 * @returns boolean
 */
mutate_error(value:mobsya.fb.ErrorType):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 6);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param number index
 * @param mobsya.fb.Breakpoint= obj
 * @returns mobsya.fb.Breakpoint
 */
breakpoints(index: number, obj?:mobsya.fb.Breakpoint):mobsya.fb.Breakpoint|null {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? (obj || new mobsya.fb.Breakpoint).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
breakpointsLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @param flatbuffers.Builder builder
 */
static startSetBreakpointsResponse(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param mobsya.fb.ErrorType error
 */
static addError(builder:flatbuffers.Builder, error:mobsya.fb.ErrorType) {
  builder.addFieldInt32(1, error, mobsya.fb.ErrorType.no_error);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset breakpointsOffset
 */
static addBreakpoints(builder:flatbuffers.Builder, breakpointsOffset:flatbuffers.Offset) {
  builder.addFieldOffset(2, breakpointsOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createBreakpointsVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startBreakpointsVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endSetBreakpointsResponse(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class WatchNode {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns WatchNode
 */
__init(i:number, bb:flatbuffers.ByteBuffer):WatchNode {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param WatchNode= obj
 * @returns WatchNode
 */
static getRootAsWatchNode(bb:flatbuffers.ByteBuffer, obj?:WatchNode):WatchNode {
  return (obj || new WatchNode).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeOrGroupId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @returns number
 */
infoType():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_info_type(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startWatchNode(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeOrGroupIdOffset
 */
static addNodeOrGroupId(builder:flatbuffers.Builder, nodeOrGroupIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeOrGroupIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number infoType
 */
static addInfoType(builder:flatbuffers.Builder, infoType:number) {
  builder.addFieldInt32(2, infoType, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endWatchNode(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class Error {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns Error
 */
__init(i:number, bb:flatbuffers.ByteBuffer):Error {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param Error= obj
 * @returns Error
 */
static getRootAsError(bb:flatbuffers.ByteBuffer, obj?:Error):Error {
  return (obj || new Error).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns mobsya.fb.ErrorType
 */
error():mobsya.fb.ErrorType {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? /**  */ (this.bb!.readInt32(this.bb_pos + offset)) : mobsya.fb.ErrorType.no_error;
};

/**
 * @param mobsya.fb.ErrorType value
 * @returns boolean
 */
mutate_error(value:mobsya.fb.ErrorType):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 6);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startError(builder:flatbuffers.Builder) {
  builder.startObject(2);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param mobsya.fb.ErrorType error
 */
static addError(builder:flatbuffers.Builder, error:mobsya.fb.ErrorType) {
  builder.addFieldInt32(1, error, mobsya.fb.ErrorType.no_error);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endError(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class RequestCompleted {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns RequestCompleted
 */
__init(i:number, bb:flatbuffers.ByteBuffer):RequestCompleted {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param RequestCompleted= obj
 * @returns RequestCompleted
 */
static getRootAsRequestCompleted(bb:flatbuffers.ByteBuffer, obj?:RequestCompleted):RequestCompleted {
  return (obj || new RequestCompleted).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startRequestCompleted(builder:flatbuffers.Builder) {
  builder.startObject(1);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endRequestCompleted(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class EventDescription {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns EventDescription
 */
__init(i:number, bb:flatbuffers.ByteBuffer):EventDescription {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param EventDescription= obj
 * @returns EventDescription
 */
static getRootAsEventDescription(bb:flatbuffers.ByteBuffer, obj?:EventDescription):EventDescription {
  return (obj || new EventDescription).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
name():string|null
name(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
name(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @returns number
 */
fixedSized():number {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_fixed_sized(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 6);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @returns number
 */
index():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_index(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 8);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startEventDescription(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nameOffset
 */
static addName(builder:flatbuffers.Builder, nameOffset:flatbuffers.Offset) {
  builder.addFieldOffset(0, nameOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number fixedSized
 */
static addFixedSized(builder:flatbuffers.Builder, fixedSized:number) {
  builder.addFieldInt32(1, fixedSized, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param number index
 */
static addIndex(builder:flatbuffers.Builder, index:number) {
  builder.addFieldInt32(2, index, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endEventDescription(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class RegisterEvents {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns RegisterEvents
 */
__init(i:number, bb:flatbuffers.ByteBuffer):RegisterEvents {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param RegisterEvents= obj
 * @returns RegisterEvents
 */
static getRootAsRegisterEvents(bb:flatbuffers.ByteBuffer, obj?:RegisterEvents):RegisterEvents {
  return (obj || new RegisterEvents).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeOrGroupId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param number index
 * @param mobsya.fb.EventDescription= obj
 * @returns mobsya.fb.EventDescription
 */
events(index: number, obj?:mobsya.fb.EventDescription):mobsya.fb.EventDescription|null {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? (obj || new mobsya.fb.EventDescription).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
eventsLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @param flatbuffers.Builder builder
 */
static startRegisterEvents(builder:flatbuffers.Builder) {
  builder.startObject(3);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeOrGroupIdOffset
 */
static addNodeOrGroupId(builder:flatbuffers.Builder, nodeOrGroupIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, nodeOrGroupIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset eventsOffset
 */
static addEvents(builder:flatbuffers.Builder, eventsOffset:flatbuffers.Offset) {
  builder.addFieldOffset(2, eventsOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createEventsVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startEventsVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endRegisterEvents(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class EventsDescriptionsChanged {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns EventsDescriptionsChanged
 */
__init(i:number, bb:flatbuffers.ByteBuffer):EventsDescriptionsChanged {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param EventsDescriptionsChanged= obj
 * @returns EventsDescriptionsChanged
 */
static getRootAsEventsDescriptionsChanged(bb:flatbuffers.ByteBuffer, obj?:EventsDescriptionsChanged):EventsDescriptionsChanged {
  return (obj || new EventsDescriptionsChanged).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeOrGroupId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param number index
 * @param mobsya.fb.EventDescription= obj
 * @returns mobsya.fb.EventDescription
 */
events(index: number, obj?:mobsya.fb.EventDescription):mobsya.fb.EventDescription|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.EventDescription).__init(this.bb!.__indirect(this.bb!.__vector(this.bb_pos + offset) + index * 4), this.bb!) : null;
};

/**
 * @returns number
 */
eventsLength():number {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.__vector_len(this.bb_pos + offset) : 0;
};

/**
 * @param flatbuffers.Builder builder
 */
static startEventsDescriptionsChanged(builder:flatbuffers.Builder) {
  builder.startObject(2);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeOrGroupIdOffset
 */
static addNodeOrGroupId(builder:flatbuffers.Builder, nodeOrGroupIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(0, nodeOrGroupIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset eventsOffset
 */
static addEvents(builder:flatbuffers.Builder, eventsOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, eventsOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param Array.<flatbuffers.Offset> data
 * @returns flatbuffers.Offset
 */
static createEventsVector(builder:flatbuffers.Builder, data:flatbuffers.Offset[]):flatbuffers.Offset {
  builder.startVector(4, data.length, 4);
  for (var i = data.length - 1; i >= 0; i--) {
    builder.addOffset(data[i]);
  }
  return builder.endVector();
};

/**
 * @param flatbuffers.Builder builder
 * @param number numElems
 */
static startEventsVector(builder:flatbuffers.Builder, numElems:number) {
  builder.startVector(4, numElems, 4);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endEventsDescriptionsChanged(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class ScratchpadUpdate {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns ScratchpadUpdate
 */
__init(i:number, bb:flatbuffers.ByteBuffer):ScratchpadUpdate {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param ScratchpadUpdate= obj
 * @returns ScratchpadUpdate
 */
static getRootAsScratchpadUpdate(bb:flatbuffers.ByteBuffer, obj?:ScratchpadUpdate):ScratchpadUpdate {
  return (obj || new ScratchpadUpdate).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns number
 */
requestId():number {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? this.bb!.readUint32(this.bb_pos + offset) : 0;
};

/**
 * @param number value
 * @returns boolean
 */
mutate_request_id(value:number):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
scratchpadId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
groupId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 8);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @param mobsya.fb.NodeId= obj
 * @returns mobsya.fb.NodeId|null
 */
nodeId(obj?:mobsya.fb.NodeId):mobsya.fb.NodeId|null {
  var offset = this.bb!.__offset(this.bb_pos, 10);
  return offset ? (obj || new mobsya.fb.NodeId).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
};

/**
 * @returns mobsya.fb.ProgrammingLanguage
 */
language():mobsya.fb.ProgrammingLanguage {
  var offset = this.bb!.__offset(this.bb_pos, 12);
  return offset ? /**  */ (this.bb!.readInt32(this.bb_pos + offset)) : mobsya.fb.ProgrammingLanguage.Aseba;
};

/**
 * @param mobsya.fb.ProgrammingLanguage value
 * @returns boolean
 */
mutate_language(value:mobsya.fb.ProgrammingLanguage):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 12);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt32(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
text():string|null
text(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
text(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 14);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @param flatbuffers.Encoding= optionalEncoding
 * @returns string|Uint8Array|null
 */
name():string|null
name(optionalEncoding:flatbuffers.Encoding):string|Uint8Array|null
name(optionalEncoding?:any):string|Uint8Array|null {
  var offset = this.bb!.__offset(this.bb_pos, 16);
  return offset ? this.bb!.__string(this.bb_pos + offset, optionalEncoding) : null;
};

/**
 * @returns boolean
 */
deleted():boolean {
  var offset = this.bb!.__offset(this.bb_pos, 18);
  return offset ? !!this.bb!.readInt8(this.bb_pos + offset) : false;
};

/**
 * @param boolean value
 * @returns boolean
 */
mutate_deleted(value:boolean):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 18);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeInt8(this.bb_pos + offset, +value);
  return true;
};

/**
 * @param flatbuffers.Builder builder
 */
static startScratchpadUpdate(builder:flatbuffers.Builder) {
  builder.startObject(8);
};

/**
 * @param flatbuffers.Builder builder
 * @param number requestId
 */
static addRequestId(builder:flatbuffers.Builder, requestId:number) {
  builder.addFieldInt32(0, requestId, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset scratchpadIdOffset
 */
static addScratchpadId(builder:flatbuffers.Builder, scratchpadIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, scratchpadIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset groupIdOffset
 */
static addGroupId(builder:flatbuffers.Builder, groupIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(2, groupIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nodeIdOffset
 */
static addNodeId(builder:flatbuffers.Builder, nodeIdOffset:flatbuffers.Offset) {
  builder.addFieldOffset(3, nodeIdOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param mobsya.fb.ProgrammingLanguage language
 */
static addLanguage(builder:flatbuffers.Builder, language:mobsya.fb.ProgrammingLanguage) {
  builder.addFieldInt32(4, language, mobsya.fb.ProgrammingLanguage.Aseba);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset textOffset
 */
static addText(builder:flatbuffers.Builder, textOffset:flatbuffers.Offset) {
  builder.addFieldOffset(5, textOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset nameOffset
 */
static addName(builder:flatbuffers.Builder, nameOffset:flatbuffers.Offset) {
  builder.addFieldOffset(6, nameOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @param boolean deleted
 */
static addDeleted(builder:flatbuffers.Builder, deleted:boolean) {
  builder.addFieldInt8(7, +deleted, +false);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endScratchpadUpdate(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  builder.requiredField(offset, 6); // scratchpad_id
  builder.requiredField(offset, 14); // text
  return offset;
};

}
}
/**
 * @constructor
 */
export namespace mobsya.fb{
export class Message {
  bb: flatbuffers.ByteBuffer|null = null;

  bb_pos:number = 0;
/**
 * @param number i
 * @param flatbuffers.ByteBuffer bb
 * @returns Message
 */
__init(i:number, bb:flatbuffers.ByteBuffer):Message {
  this.bb_pos = i;
  this.bb = bb;
  return this;
};

/**
 * @param flatbuffers.ByteBuffer bb
 * @param Message= obj
 * @returns Message
 */
static getRootAsMessage(bb:flatbuffers.ByteBuffer, obj?:Message):Message {
  return (obj || new Message).__init(bb.readInt32(bb.position()) + bb.position(), bb);
};

/**
 * @returns mobsya.fb.AnyMessage
 */
messageType():mobsya.fb.AnyMessage {
  var offset = this.bb!.__offset(this.bb_pos, 4);
  return offset ? /**  */ (this.bb!.readUint8(this.bb_pos + offset)) : mobsya.fb.AnyMessage.NONE;
};

/**
 * @param mobsya.fb.AnyMessage value
 * @returns boolean
 */
mutate_message_type(value:mobsya.fb.AnyMessage):boolean {
  var offset = this.bb!.__offset(this.bb_pos, 4);

  if (offset === 0) {
    return false;
  }

  this.bb!.writeUint8(this.bb_pos + offset, value);
  return true;
};

/**
 * @param flatbuffers.Table obj
 * @returns ?flatbuffers.Table
 */
message<T extends flatbuffers.Table>(obj:T):T|null {
  var offset = this.bb!.__offset(this.bb_pos, 6);
  return offset ? this.bb!.__union(obj, this.bb_pos + offset) : null;
};

/**
 * @param flatbuffers.Builder builder
 */
static startMessage(builder:flatbuffers.Builder) {
  builder.startObject(2);
};

/**
 * @param flatbuffers.Builder builder
 * @param mobsya.fb.AnyMessage messageType
 */
static addMessageType(builder:flatbuffers.Builder, messageType:mobsya.fb.AnyMessage) {
  builder.addFieldInt8(0, messageType, mobsya.fb.AnyMessage.NONE);
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset messageOffset
 */
static addMessage(builder:flatbuffers.Builder, messageOffset:flatbuffers.Offset) {
  builder.addFieldOffset(1, messageOffset, 0);
};

/**
 * @param flatbuffers.Builder builder
 * @returns flatbuffers.Offset
 */
static endMessage(builder:flatbuffers.Builder):flatbuffers.Offset {
  var offset = builder.endObject();
  return offset;
};

/**
 * @param flatbuffers.Builder builder
 * @param flatbuffers.Offset offset
 */
static finishMessageBuffer(builder:flatbuffers.Builder, offset:flatbuffers.Offset) {
  builder.finish(offset);
};

}
}
